firefox-ios/Client/Frontend/Autofill/Address/Edit/EditAddressWebViewManager.swift

```diff
@@ -44,4 +50,16 @@ class EditAddressWebViewManager: WebViewPreloadManaging {
         webView?.removeFromSuperview()
         webView = nil
     }
+
+    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
+        guard message.name == "saveEnabled",
+              let body = message.body as? [String: Any],
+              let saveEnabled = body["enabled"] as? Bool else { return }
+
+        NotificationCenter.default.post(name: .addressSettingsSaving, object: nil, userInfo: ["enabled": saveEnabled])
```

Comment:
@nbhasin2 @razvanlitianu I used the `NotificationCenter` here as it's the easiest way. Not sure if this is the best approach. An alternative is passing the viewModel to this class. What do you think ?

Reply:
Hey Issam, there are a couple of options. One can be NotificationBased but it's pretty loose in the sense that everyone can respond to it. There is also delegation or closure based sending.

Reply:
Delegation or closure is usually the best approach to share data unless you want to notify everyone

@issammani think of this way

Delegation is 1 to 1 subscription using protocols
Closure is also 1 to 1 but its syntactically different than how delegation works however makes passing data easy.
Notification is technically 1 to many so anyone can subscribe to it 

For your case closure seems like the way to go but if you feel its tedious then you can fall back to classic delegation

Reply:
Thanks @nbhasin2 @razvanlitianu for the thorough explaination. I will opt in the closure based approach since it seems more appropriate here 😄